### PR TITLE
fix default fetch function type

### PR DESCRIPTION
**Pull request by mnaray**
I created a new ts interface for the json that comes in from the backend. This way react's ts-processor can deconstruct it and handle it like an object with properties. This makes it possible for the react node to display it onto the screen as the type is compatible

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -2,17 +2,25 @@ import React, { useEffect, useState } from 'react';
...
+
+  const [backendData, setBackendData] = React.useState<IApiData | null>(null);
 
   useEffect(() => {
     fetch("/api").then(
       response => response.json()
     ).then(
       data => {
         setBackendData(data)
+        console.log(backendData)
```
Comment (sanqro) [inline client/src/App.tsx:23]: Please remove this line of code. It's bloat :sweat_smile:

**Review by sanqro — CHANGES_REQUESTED** [review]
Please have a look at my comments :+1:

**Review by sanqro — APPROVED** [review]
Your code seems fine now :+1:

File: client/src/App.tsx [hunk 2]
```diff
@@ -17,13 +24,19 @@ function App() {
...
+  }
+
   return (
     <div className="App">
       <header className="App-header">
         <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          GNULet's go! 😎
-        </p>
+        <p>{checkIfFetched(backendData)}</p>
```
Comment (RelxOff) [inline client/src/App.tsx:39]: nice short fix 🙂

**Review by RelxOff — APPROVED** [review]
seems fine 👍

**Pull request merged**